tests/utils/test_bump_version.py

```diff
@@ -0,0 +1,75 @@
+import pytest
+import semver
+
+from tests.copilot_helper.conftest import BASE_DIR
+from utils.bump_version import bump_version_if_required
+from utils.bump_version import get_pyproject_version
+from utils.bump_version import version_should_be_bumped
+
+PYPROJECT_CONTENT = """
+[tool.poetry]
+name = "dbt-copilot-tools"
+version = "%s"
+"""
+
+
+def test_get_pyproject_version(fs):
+    fs.create_file(f"{BASE_DIR}/pyproject.toml", contents=PYPROJECT_CONTENT % "1.2.3")
+    version = get_pyproject_version()
+
+    assert version == "1.2.3"
+
+
+@pytest.mark.parametrize(
+    "version, bump_version, expected_version,expected_return_code",
+    [
+        ("1.2.3", True, "1.2.5", 1),
+        ("1.2.3", False, "1.2.4", 1),
+        ("1.2.4", True, "1.2.5", 1),
+        ("1.2.4", False, "1.2.4", 0),
+        ("1.3.0", True, "1.3.0", 0),
+        ("1.3.0", False, "1.3.0", 0),
+    ],
+)
+def test_bump_version_if_required(
+    version, bump_version, expected_version, expected_return_code, fs, capsys
+):
+    versions = ["1.1.1", "1.1.2", "1.2.3", "1.2.4"]
```

Comment:
Just trying to understand what's going on here. Is `versions` actually `existing_published_versions`?

Reply:
Yes. I'll have a think about renaming to be more explicit

Reply:
Oh - actually read existing_published_versions without the underscores. Changed to something similar though